runtime/interpreter/decode.go

```diff
@@ -1755,6 +1762,77 @@ func (d *DecoderV4) decodeArrayElements(path []string) ([]Value, error) {
 	return values, nil
 }
 
+func (d *DecoderV4) decodeArrayValueStaticType(valuePath []string) (StaticType, error) {
+	const expectedLength = encodedArrayValueLength
+
+	size, err := d.decoder.DecodeArrayHead()
+
+	if err != nil {
+		if e, ok := err.(*cbor.WrongTypeError); ok {
+			return nil, fmt.Errorf("invalid array encoding (@ %s): expected [%d]interface{}, got %s",
+				strings.Join(valuePath, "."),
+				expectedLength,
+				e.ActualType.String(),
+			)
+		}
+		return nil, err
+	}
+
+	if size != expectedLength {
+		return nil, fmt.Errorf("invalid array encoding (@ %s): expected [%d]interface{}, got [%d]interface{}",
+			strings.Join(valuePath, "."),
+			expectedLength,
+			size,
+		)
+	}
+
+	// Static type
+
+	// Decode type at array index encodedArrayValueStaticTypeFieldKey
+	return d.decodeStaticType()
+}
+
+func decodeArrayMetaInfo(array *ArrayValue, content []byte) error {
+	d, err := NewByteDecoder(content, array.Owner, array.encodingVersion, array.decodeCallback)
+	if err != nil {
+		return err
+	}
+
+	// TODO: store array type
+	//   Option 1: convert to sema type. - Don't have the interpreter
+	//   Option 2: Store static type in array
+	_, err = d.decodeArrayValueStaticType(array.valuePath)
```

Comment:
Like discussed, I think it might be enough to go with Option 2, i.e. only have the static type in the array value, and then convert it to the sema type when needed / where the interpreter is available. That way we don't have to add the interpreter to the decoder here.